insights/combiners/tests/test_cloud_provider.py

```diff
@@ -0,0 +1,422 @@
+Chassis Information
+\tManufacturer: Xen
+\tType: Other
+\tLock: Not Present
+\tVersion: Not Specified
+\tSerial Number: Not Specified
+\tAsset Tag: Not Specified
+\tBoot-up State: Safe
+\tPower Supply State: Safe
+\tThermal State: Safe
+\tSecurity Status: Unknown
+'''
+
+DMIDECODE_GOOGLE = '''
+# dmidecode 2.12-dmifs
+SMBIOS 2.4 present.
+11 structures occupying 310 bytes.
+Table at 0x000EB01F.
+
+Handle 0x0000, DMI type 0, 24 bytes
+BIOS Information
+\tVendor: Google
+\tVersion: Google
+\tRelease Date: 12/09/2016
+\tAddress: 0xE8000
+\tRuntime Size: 96 kB
+\tROM Size: 64 kB
+\tCharacteristics:
+\t\tPCI is supported
+\t\tEDD is supported
+\t\tTargeted content distribution is supported
+\tBIOS Revision: 4.2
+
+Handle 0x0100, DMI type 1, 27 bytes
+System Information
+\tManufacturer: Xen
+\tProduct Name: HVM domU
+\tVersion: 4.2.amazon
+\tSerial Number: ec2f58af-2dad-c57e-88c0-a81cb6084290
+\tUUID: EC2F58AF-2DAD-C57E-88C0-A81CB6084290
+\tWake-up Type: Power Switch
+\tSKU Number: Not Specified
+\tFamily: Not Specified
+
+Handle 0x0300, DMI type 3, 13 bytes
+Chassis Information
+\tManufacturer: Xen
+\tType: Other
+\tLock: Not Present
+\tVersion: Not Specified
+\tSerial Number: Not Specified
+\tAsset Tag: Not Specified
+\tBoot-up State: Safe
+\tPower Supply State: Safe
+\tThermal State: Safe
+\tSecurity Status: Unknown
+'''
+
+DMIDECODE_AZURE_ASSET_TAG = """
+# dmidecode 3.1
+Getting SMBIOS data from sysfs.
+SMBIOS 2.8 present.
+10 structures occupying 511 bytes.
+Table at 0x000F6050.
+
+Handle 0x0000, DMI type 0, 24 bytes
+BIOS Information
+\tVendor: SeaBIOS
+\tVersion: 1.11.0-2.el7
+\tRelease Date: 04/01/2014
+\tAddress: 0xE8000
+\tRuntime Size: 96 kB
+\tROM Size: 64 kB
+\tCharacteristics:
+\t\tBIOS characteristics not supported
+\t\tTargeted content distribution is supported
+\tBIOS Revision: 0.0
+
+Handle 0x0100, DMI type 1, 27 bytes
+System Information
+\tManufacturer: oVirt
+\tProduct Name: oVirt Node
+\tVersion: 7-5.1804.4.el7.centos
+\tSerial Number: 30393137-3436-584D-5136-323830304E46
+\tUUID: a35ae32b-ed0a-49a4-9dbb-eecf21f88aab
+\tWake-up Type: Power Switch
+\tSKU Number: Not Specified
+\tFamily: Red Hat Enterprise Linux
+
+Handle 0x0300, DMI type 3, 21 bytes
+Chassis Information
+\tManufacturer: Red Hat
+\tType: Other
+\tLock: Not Present
+\tVersion: RHEL 7.2.0 PC (i440FX + PIIX, 1996)
+\tSerial Number: Not Specified
+\tAsset Tag: 7783-7084-3265-9085-8269-3286-77
+\tBoot-up State: Safe
+\tPower Supply State: Safe
+\tThermal State: Safe
+\tSecurity Status: Unknown
+\tOEM Information: 0x00000000
+\tHeight: Unspecified
+\tNumber Of Power Cords: Unspecified
+\tContained Elements: 0
+"""
+
+DMIDECODE_FAIL = "# dmidecode 2.11\n# No SMBIOS nor DMI entry point found, sorry.\n"
+
+
+def test_rpm_google():
+    irpms = IRPMS(context_wrap(RPMS_GOOGLE))
+    dmi = DMIDecode(context_wrap(DMIDECODE))
+    yrl = YumRepoList(context_wrap(YUM_REPOLIST_NOT_AZURE))
+    ret = CloudProvider(irpms, dmi, yrl)
+    assert ret.cloud_provider is 'google'
```

Comment:
Use `==` for equality between strings. `is` does an address comparison, which means it might work or not depending on whether the string is in the internal cache.

Reply:
Thanks for the input Chris, made suggested changes.